gh-133438: Fix the use of the terms "argument" and "parameter" in the documentation (#135160)
python/cpython: Doc/c-api/arg.rst

``$``
   :c:func:`PyArg_ParseTupleAndKeywords` only:
   Indicates that the remaining arguments in the Python argument list are

>    keyword-only.  Currently, all keyword-only arguments must also be optional

Reviewer: I think the old wording is clearer here, "keyword-only" is a well-known term in Python.

Author: Keyword-only can be only parameters. Arguments are either positional or keyword. Argument is what we pass to the function.

Reviewer 2: In the same way, an argument can't be *optional*: if you omit it, it does not exist at all.

But the terms “optional” and “keyword-only” describe the possibilities you have, not a specific argument in a specific call.

>    supplied only by keyword.  Currently, all such arguments must also be optional

Reviewer 2: Sadly, “keyword” means something entirely different. IMO we need to use the full term.
```suggestion
   supplied only as keyword arguments.  Currently, all such arguments must also be optional
```

Author: This is from the definition of the keyword-only parameter.
```rst
      * :dfn:`keyword-only`: specifies an argument that can be supplied only
        by keyword.  Keyword-only parameters can be defined by including a
```

Reviewer 3: In the context of `PyArg_ParseTupleAndKeywords`, 'keyword' means 'parameter name' rather than 'syntax keyword.  So I think Serhiy's revision could be left as is, or changed to
```suggestion
   supplied only by parameter name.  Currently, all such arguments must also be optional
```
The `keword-only` definition is a bit circular.  Perhaps `by keyword` should be changed to `by parameter name` to break the loop.

.. c:function:: int PyArg_Parse(PyObject *args, const char *format, ...)

>    Parse the argument of a function that takes a single positional parameter

Reviewer: ```suggestion
   Parse the argument of a function that takes a single positional argument
```
